spanreceiver/spanreceiver.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanreceiver
+
+import (
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+// SpanReceiver is an interface that receives spans from a Node identifier.
+type SpanReceiver interface {
+	ReceiveSpans(node *commonpb.Node, spans ...*tracepb.Span) (*Acknowledgement, error)
```

Comment:
Not a go expert, but what is the advantage to use variadic here vs an array?

Reply:
Great question! It makes for a nice API usage when calling items individually
```go
ReceiveSpans(node, span1, span2)
```
vs requiring users to create an array
```go
spans := []*tracepb.Span{span1, span2}
ReceiveSpans(node, spans)
// or
ReceiveSpans(node, []*tracepb.Span{span1, span2})
```
but also works in the general case of when we have an array
```go
ReceiveSpans(node, spans...)
```

Reply:
Great question! It makes for a nice API usage when calling items individually
```go
ReceiveSpans(node, span1, span2)
```
vs requiring users to create a list first
```go
spans := []*tracepb.Span{span1, span2}
ReceiveSpans(node, spans)
// or
ReceiveSpans(node, []*tracepb.Span{span1, span2})
```
but also works in the general case of when we have a list
```go
ReceiveSpans(node, spans...)
```